### DISCUSSION TITLE
Firebase Auth and Vendure

### DISCUSSION THREAD

**Discussion by rohanrajpal** [vendurehq/vendure#667] [Q&A]
Hey, I am using firebase auth with Vendure and was trying to setup session cookies

https://firebase.google.com/docs/auth/admin/manage-cookies#node.js

I am having some trouble creating the cookie from the [server side](https://firebase.google.com/docs/auth/admin/manage-cookies#create_session_cookie), as from the [vendure authentication](https://www.vendure.io/docs/developer-guide/authentication/) docs, the authenticate functions only return the user and I couldn't find a way to modify session cookies. 

so how do I modify the cookie when I verify the id and csrf token in `FirebaseAuthStrategy.tsx`?

**Comment by michaelbromley** [comment]
Hi Rohan,

I'm not familiar with Firebase auth, but in the example I see they set the cookie using `res.cookie('session', sessionCookie, options);`. In Vendure there is no `cookie` method on the response object, but looking at [this firebase example repo](https://github.com/firebase/quickstart-nodejs/blob/master/auth-sessions/app.js#L121) it looks like they are using the `cookie-parser` middleware to enable this.

So you could try installing that package, adding it to your `apiOptions.middleware` array, and then in your auth handler, check `ctx.req.res` to see if there is now a `cookie()` method.

**Reply by rohanrajpal** [reply]
Hey, thanks a lot. Let me try this out.

**Reply by kevmtt** [reply]
Hey @rohanrajpal let me know if you made progress on that topic. I'm also thinking about using FirebaseAuth or AWS Amplify Authentication :)
Cheers

**Reply by rohanrajpal** [reply]
Hey @Draykee 

Yeah, I just started working on this again. @michaelbromley I had a few doubts

Is this the right way to enable cookie-parser middleware? I'm not able to get the cookie method in the response headers to work

```ts
const cookieParser = require("cookie-parser");
export const config: VendureConfig = {
...
    middleware: [
      {
        handler: cookieParser,
        route: "admin",
      },
    ],
...
  },
```

Also if we set the cookies on ourselves, what would happen the `set-cookie` that is returned by `authenticateUser`?

**Reply by michaelbromley** [reply]
Not sure if you need to call the `cookieParser()` function - try it and see if that works.

> Also if we set the cookies on ourselves, what would happen the set-cookie that is returned by authenticateUser?

I think you can set multiple if you give yours a different name (Vendure uses the name "session" by default).

**Reply by rohanrajpal** [reply]
the `cookieParser()` function also does not work :(

Yeah I was thinking multiple cookie names too.

**Reply by michaelbromley** [reply]
You can also try using the [beforeVendureBootstrap](https://www.vendure.io/docs/typescript-api/plugin/plugin-lifecycle-methods/#beforevendurebootstrap) plugin lifecycle hook to add the middleware, as it gives you access to the Express `app` object.

```ts
class MyPlugin {
  static beforeVendureBootstrap(app: INestApplication) {
    app.use(cookieParser());
  }
}
```

**Reply by rohanrajpal** [reply]
Yeah tried that too. Didn't work :(
Let me try cookie parser in a minimal project and understand how it works. I think I'm using it incorrectly.
